queries/__init__.py

```diff
@@ -253,6 +253,546 @@ def convert(m):
+        if number == "ft" or (gender == "kvk" and decl != "nf"):
+            word = "þúsundustu"
+        else:
+            word = "þúsund" + _LARGE_ORDINAL_SUFFIX[gender][decl]
+
+    elif "jón" in word:
+        if number == "ft":
+            word = re.sub(r"(\S*jón)\S*", r"\1ustu", word)
+        else:
+            word = re.sub(
+                r"(\S*jón)\S*", r"\1" + _LARGE_ORDINAL_SUFFIX[gender][decl], word
+            )
+
+    elif "jarð" in word:
+        if number == "ft" or (gender == "kvk" and decl != "nf"):
+            word = re.sub(r"(\S*)jarð\S*", r"\1jörðustu", word)
+        else:
+            word = re.sub(
+                r"(\S*jarð)\S*", r"\1" + _LARGE_ORDINAL_SUFFIX[gender][decl], word
+            )
+
+    return word
+
+
+def neutral_text_to_ordinal(
+    s: str, decl: str = "nf", gender: str = "kk", number: str = "et"
+) -> str:
+    """
+    Takes Icelandic text representation of number
+    and returns it as an ordinal in specified declension (nf, þf, þgf, ef),
+    gender (kk, kvk, hk) and number (et, ft).
+    """
+    if len(s) <= 0:
+        return s
+
+    ordinal: List[str] = s.split()
+
+    # Change last word to ordinal
+    ordinal[-1] = _num_to_ordinal(ordinal[-1], decl, gender, number)
+
+    if len(ordinal) > 1:
+        # Change e.g. "tvö þúsund og fyrsti" -> "tvö þúsundasti og fyrsti"
+        if ordinal[-2] == "og" and len(ordinal) >= 3:
+            # Check that last number in text isn't a large ordinal
+            # e.g. "sextugustu", "hundraðasti" or "þúsundasta"
+            if not re.search(r"[au]st[iau]$", ordinal[-1]):
+                ordinal[-3] = _num_to_ordinal(ordinal[-3], decl, gender, number)
+
+    ordinal_str: str = " ".join(ordinal)
+
+    # Change e.g.
+    # "eitt hundraðasti" -> "hundraðasti"
+    # "ein milljónasta og fyrsta" -> "milljónasta og fyrsta"
+    ordinal_str = re.sub(r"^(einn?|eitt) ((\S*)([au]st[iau]))", r"\2", ordinal_str)
+
+    return ordinal_str
+
+
+def number_to_ordinal(
+    n: int, decl: str = "nf", gender: str = "kk", number: str = "et"
+) -> str:
+    """
+    Takes number and returns it as an ordinal
+    in specified declension (nf, þf, þgf, ef),
+    gender (kk, kvk, hk) and number (et, ft).
+    """
+    return neutral_text_to_ordinal(number_to_neutral(n), decl, gender, number)
+
+
+_DIGITS_TO_KK = {
+    "0": "núll",
+    "1": "einn",
+    "2": "tveir",
+    "3": "þrír",
+    "4": "fjórir",
+    "5": "fimm",
+    "6": "sex",
+    "7": "sjö",
+    "8": "átta",
+    "9": "níu",
+}
+
+
+def digits_to_text(digit_list: Iterable[Union[str,int]]) -> str:
+    """
+    Takes in a string of digits (or list of strings/ints)
+    and returns as spoken text in Icelandic.
+    Useful for phone numbers, social security numbers and such.
+    Examples:
+        "5885522" -> "fimm átta átta fimm fimm tveir tveir"
+        ["234",1,"9"] -> "tveir þrír fjórir einn níu"
+    """
+    digit_text: List[str] = []
+
+    for d in digit_list:
+        d = str(d).strip()
```

Comment:
Could do `if not d: continue` here, and then skip both length checks below

Reply:
Good idea! Fixed